tests/unit/test__types.py

```diff
@@ -0,0 +1,232 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+
+from sqlalchemy_bigquery._types import _get_transitive_schema_fields
+from google.cloud.bigquery.schema import SchemaField
+
+
+def create_schema_field_from_dict(schema_dict):
+    """
+    Helper function to create a SchemaField object from a dictionary representation.
+    """
+    api_repr = {
+        "name": schema_dict["name"],
+        "type": schema_dict["type"],
+        "mode": schema_dict.get("mode", "NULLABLE"),
+        "fields": [
+            create_schema_field_from_dict(sf_dict).to_api_repr()
+            for sf_dict in schema_dict.get("fields", [])
+        ],
+    }
+    return SchemaField.from_api_repr(api_repr)
+
+
+test_cases = [
+    (
+        "STRUCT field, not REPEATED, with sub-fields, should recurse",
```

Comment:
[Optional] Generally, I use [pytest.param()](https://docs.pytest.org/en/latest/reference/reference.html#pytest-param) with an ID argument instead of a description. That makes it a little easier to select a particular test. See 

https://github.com/googleapis/python-bigquery-dataframes/blob/08d70b6ad3ab3ac7b9a57d93da00168a8de7df9a/tests/system/small/operations/test_plotting.py#L203-L204

for an example.

Reply:
Good point, thanks for the reminder on this option.